### PR TITLE
RATIS-1702. Make listener reject leader election

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1702

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/VoteContext.java [hunk 1]
```diff
@@ -131,6 +132,9 @@ RaftPeer recognizeCandidate(long candidateTerm) {
    * See Section 5.4.1 Election restriction
    */
   boolean decideVote(RaftPeer candidate, TermIndex candidateLastEntry) {
+    if (impl.getRole().getCurrentRole() == RaftProtos.RaftPeerRole.LISTENER) {
+      return reject("The server " + impl.getId() + " is listener and should not participate in elections");
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/VoteContext.java:136]: @codings-dan , thanks a lot for working on this!  The id is already printed inside `reject(..)`.  How about changing the message to the following?
```java
 return reject("this server is a listener, who is a non-voting member");
```

Comment (codings-dan) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/VoteContext.java]: Good catch! I have change the message.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by codings-dan** [comment]
Thanks @szetszwo for helping review the pull request!

**Pull request merged**